Provider/FileProvider.php

```diff
@@ -169,27 +173,27 @@ public function postUpdate(MediaInterface $media)
         $this->generateThumbnails($media);
 
         $media->resetBinaryContent();
+
+        $this->removeTmpFiles($media);
     }
 
     /**
-     * @throws \RuntimeException
-     *
      * @param \Sonata\MediaBundle\Model\MediaInterface $media
-     *
-     * @return
      */
     protected function fixBinaryContent(MediaInterface $media)
     {
-        if ($media->getBinaryContent() === null) {
+        if ($media->getBinaryContent() === null || $media->getBinaryContent() instanceof File) {
             return;
         }
 
-        // if the binary content is a filename => convert to a valid File
-        if (!$media->getBinaryContent() instanceof File) {
-            if (!is_file($media->getBinaryContent())) {
-                throw new \RuntimeException('The file does not exist : ' . $media->getBinaryContent());
-            }
+        if ($media->getBinaryContent() instanceof Request) {
+            $this->generateBinaryFromRequest($media);
+
+            return;
+        }
 
+        // if the binary content is a filename => convert to a valid File
+        if (is_file($media->getBinaryContent())) {
```

Comment:
You still need to throw the exception if the binary content is not a file afterwards, don't you?